CIScripts/BuildStage.ps1

```diff
@@ -23,13 +27,21 @@ if ($NothingToBuild -or $CopyDisabledArtifacts) {
     if (-Not (Test-Path "$OutputRootDirectory")) {
         New-Item -Name $OutputRootDirectory -ItemType directory
     }
-    Copy-Item ("$DiskName" + ":\*") -Destination "$OutputRootDirectory\" -Recurse -Container
 
-    $Job.Done()
-}
+    foreach ($Item in Get-ChildItem "${DiskName}:\") {
+        $OutputItem = "$OutputRootDirectory\$($Item.Name)"
 
-if (-not $NothingToBuild) {
-    & $PSScriptRoot\Build.ps1
+        if ((Test-Path $OutputItem) -And (Get-ChildItem $OutputItem | Measure-Object).Count -ne 0) {
+            # Skip existing files and non-empty directories
```

Comment:
This comment wouldn't be necessary if we named the elements of `if` statement. Something like

```
$FileAlreadyExists = Test-Path $OutputItem
$IsNonEmptyDirectory = (Get-ChildItem $OutputItem | Measure-Object).Count -ne 0
if($FileAlreadyExists -Or $IsNonEmptyDirectory)
{ continue }
```

Also, shouldn't there be an -Or instead of -And?

Reply:
We can't extract the conditions to variables because lhs serves as guard for rhs.

And yes, `and` is the correct behaviour here. With `or` we'd skip empty directories too.

Anyway, I have an idea how to simplify this condition, so stay tuned.

Reply:
That's what I suspected but didn't really read into it since I decided it's not readable enough as it is. I think we should avoid if statements that do anything more than check a condition if possible.